lib/addons/cert-manager/index.ts

```diff
@@ -0,0 +1,76 @@
+// lib/certmanager_addon.ts
+import { Construct } from 'constructs';
+import merge from "ts-deepmerge";
+import { setPath } from '../../utils/object-utils';
+import { HelmAddOn, HelmAddOnUserProps, HelmAddOnProps } from "../helm-addon";
+import { ClusterInfo, Values } from "../../spi";
+import { createNamespace } from "../../utils";
+/**
+ * User provided options for the Helm Chart
+ */
+export interface CertManagerAddOnProps extends HelmAddOnUserProps {
+    /**
+     * Version of the helm chart to deploy
+     */      
+    version?: string;
+    /**
+     * To automatically install and manage the CRDs as part of your Helm release,
+     */    
+    installCRDs?: boolean;
+    /**
+     * To Create Namespace using CDK
+     */    
+    createNamespace?: boolean;
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & CertManagerAddOnProps = {
+  name: "blueprints-cert-manager-addon",
+  namespace: "cert-manager",
+  chart: "cert-manager",
+  version: "1.8",
+  release: "cert-manager",
+  repository:  "https://charts.jetstack.io",
+  values: {},
+  installCRDs: true, //To automatically install and manage the CRDs as part of your Helm release,
+  createNamespace: true
+
+};
+
+/**
+ * Main class to instantiate the Helm chart
+ */
+export class CertManagerAddOn extends HelmAddOn {
+
+  readonly options: CertManagerAddOnProps;
+
+  constructor(props?: CertManagerAddOnProps) {
+    super({...defaultProps, ...props});
+    this.options = this.props as CertManagerAddOnProps;
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const cluster = clusterInfo.cluster;
+    
+    if( this.options.createNamespace == true){
+      createNamespace(this.options.namespace! , cluster);
+    }
+    
+    let values: Values = populateValues(this.options);
+    values = merge(values, this.props.values ?? {});
+    const chart = this.addHelmChart(clusterInfo, values);
+    return Promise.resolve(chart);

```

Comment:
Cert manager stand alone can issue certificates from a variety of supported sources, including [Let's Encrypt](https://letsencrypt.org/), [HashiCorp Vault](https://www.vaultproject.io/), and [Venafi](https://www.venafi.com/) as well as private PKI.

For integrating cert manager with AWS ACM PCA we need to use another addon - aws-privateca-issuer
https://github.com/cert-manager/aws-privateca-issuer
Even though it is managed by certmanager , its altogether a separate helm package and have separate lifecycle and hence i wanted to build a new addon for this

I am already in the process of creating this ] addon for aws-privateca-issuer and I will raise pull request once cert manager is approved ( since it has dependency on cert manager I am planning to use @dependable decorator)

Let me know if it makes sense?

Reply:
Yes, makes perfect sense.